### ISSUE TITLE
Need to double check that Size of Arrows and Effort of Robots (cancel out)

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#182]
@zepumph, apologies for opening this issue so late. I do think there was an issue somewhere between GFL regular, GFLB and ISLC where I pointed out that the effort of robots needs to line up with the description of the size of the arrows.

I can't seem to find the issue where I posted the table, so I am posting it again.

Size of force arrows | Effort of the robots
-- | --
tiny | pull very little
very small | pull a little
small | pull lightly
medium size | pull gently
large | pull moderately
very large | pull hard
huge | pull very hard

There seems to be problem when the force arrows are huge and off the screen. They stay huge and off the screen for a while, but when decreasing the mass the robots animate to lessen their effort and the describe effort decreases even when the size of the arrows do not.

@arouinfar, can we discuss this discrepancy? There is a big difference in force value, so maybe it is fine as-is. 
<img width="1111" alt="Screen Shot 2019-10-02 at 3 24 25 PM" src="https://user-images.githubusercontent.com/1902116/66068828-e2522e80-e528-11e9-9275-0503f4171bac.png">

<img width="1114" alt="Screen Shot 2019-10-02 at 3 24 57 PM" src="https://user-images.githubusercontent.com/1902116/66068845-e7af7900-e528-11e9-9230-d55ff2a8bf7a.png">

**Comment by zepumph** [comment]
This was due to the model bug found in https://github.com/phetsims/inverse-square-law-common/issues/76. Basically the range that these description values were based on doubled. I can look into fixing those values.

**Comment by zepumph** [comment]
(this will apply to all ISL sims)

**Comment by terracoda** [comment]
Thanks @zepumph for connecting the correct issue into this one.
The descriptions are aligned on sim start-up.
![Screen Shot 2019-10-02 at 5 51 30 PM](https://user-images.githubusercontent.com/1902116/66078855-6adaca00-e53d-11e9-84a8-d90e6ad23913.png)

I'll look at this more tomorrow.

**Comment by terracoda** [comment]
@arouinfar, I think the main problem is that the arrows are shrinking, but that shrinking is still off screen, so you only see the change in the robots. Maybe the arrows can be described as very large or huge and be visually off screen?

Let em know your thoughts @arouinfar

**Comment by zepumph** [comment]
Sorry I keep barging in here, but want to make sure that the different descriptions are being appropriately understood as they apply to the force. All forces are in newtons and from GFLB.

* The force range for the sim is ~.7 to ~3405. Min is smallest mass, farthest away. Max is largest mass constant size, closest to each other. 
* force to pull effort "region" is mapped linearly, from the min force to half the max (~1700)
* Arrow size regions are determined from a custom table that looks like this (where 0 is tiny, and 6 is huge):
  ```js
 
      if ( force <= 85.4 ) {
        return 0;
      }
      if ( force <= 181.6 ) {
        return 1;
      }
      if ( force <= 278.1 ) {
        return 2;
      }
      if ( force <= 386.1 ) {
        return 3;
      }
      if ( force <= 486.1 ) {
        return 4;
      }
      if ( force <= 576.6 ) {
        return 5;
      }
      return 6;
  ```

All of these feel quite arbitrary, and perhaps the manula force mapping for the arrow strings were calculated thinking that the max was ~1098, where the constant size isn't checked, by mass is greatest and distance shortest.

Let me know if this is helpful. I'm also happy to discuss further. 

When we fixed the bug in https://github.com/phetsims/inverse-square-law-common/issues/76, I divided the linear function for the puller strings by 2, but didn't change the arrow region values.

**Comment by terracoda** [comment]
This is very helpful and it seems we may have tired some description hollywooding based on the old force values.

I will look at this more tomorrow, and no worries about barging in.

It seems if we want the descriptions to match up (or cancel out so-to-speak) for force and counter force, then we should be calculating them in the same way.

**Comment by zepumph** [comment]
Furthermore, for the visuals both use linear functions, 
* the pullers are mapped from the min for to the max force directly
* the arrows are mapped from the min magnitude (so they can't be negative in coulombs law), to the max force. 

Both of these maxes are that 3405 value where they used to be 1098.

**Comment by arouinfar** [comment]
@terracoda I think it's important to keep the force arrow and effort descriptions parallel -- `tiny` always corresponds to `pull very little`, etc. The robot's pull is directly proportional to the force between the masses.

I'm not sure what range of force each size/effort description level corresponds to. However, I think the range needs to be equal for each level, save for the `huge`/`pull very hard` category, which could be tied to the point at which the vectors are no longer fully visible.

**Comment by terracoda** [comment]
Thanks @arouinfar! That is what I thought, too.

@zepumph, I think what you will need from me is a new set of ranges for the 7 described sizes/efforts. The new set would start at the min force value, ~.7, and end at ~3405, the max force value.

If my understading is correct, the visuals are animated to map to this range in the desired way.

For me to map in the descriptions I need to play with the current version to and decide on the new range for tiny/pull very little, etc.

**Comment by terracoda** [comment]
@zepumph and I had a zoom call, and decided I will make a new value set for the ranges tomorrow morning, and he already made it so the size of the arrows and the effort of the robots will be mapped to the new value sets that I will come up with.

**Comment by zepumph** [comment]
>and he already made it so the size of the arrows and the effort of the robots will be mapped to the new value sets that I will come up with.

Now I did!

**Comment by terracoda** [comment]
@zepumph, apologies, I am late on this, making it priority now.

**Comment by terracoda** [comment]
Size of Force Arrows | Current range values (approx. N range) | Effort  of Robots
-- | --
Tiny | 0.7 - 85.4(84.7 N)=< 40.6 N |
Very small | 92.7 - 181.6(88.9 N +gap) 40.7 - =< 250.3( 209.6 N) |
Small | 200.2 -  278.1(77.9 N +gap) 250.4 - =< 544.8( 294.6 N) |
Medium size | 295.8 - 386.1(90.3 N +gap)544.9 - =< 888.6(343.7 N) |
Large | 400.4 - 486.1(85.7 N +gap)888.7 - =< 1261.6(372.9 N) |
Very large | 493.6 -  576.6(83.0 +gap)1261.7 - =< 2059.9(798.8 N) |
Huge | 592.4 - 1067.8(475.4 N)2060 - =< 3949.2(1189.2 N) |

**Comment by terracoda** [comment]
@zepumph, I played a lot with the sim and I hope these ranges will work out.
I am not sure I was supposed to use the max value for each `if` clause.

Back over to you.

**Comment by zepumph** [comment]
The ranges you gave are perfect. I'll put those in now thanks.

**Comment by terracoda** [comment]
@zepumph, I need to do something similar for GFL regular, too right?
I'll do that over in https://github.com/phetsims/gravity-force-lab/issues/178

And then we can discuss new strategies for this type of thing in 
https://github.com/phetsims/a11y-research/issues/146

Apologies for confusing all these issues.

**Comment by zepumph** [comment]
There was a discrepancy between ISLCModel.getMaxForce() and the highest value you could get in the sim. This was because the model function wasn't snapping to the grid of actual possibilities. Fixed https://github.com/phetsims/inverse-square-law-common/commit/c95130a2ea7817efccd25ff397df82824594ecca.

I also input the regions, and saw that they were changing well in the a11y view. I will move these over to the branch now:

ISLC:
https://github.com/phetsims/inverse-square-law-common/commit/4999161959adaedbb42d63c1851b9da86a1f432d
https://github.com/phetsims/inverse-square-law-common/commit/c95130a2ea7817efccd25ff397df82824594ecca

GFLB:
https://github.com/phetsims/gravity-force-lab-basics/commit/61c089911cb99bdd2f20c084287d5bc7d2a58cc7

**Comment by zepumph** [comment]
QA, to test, please make sure that the force description regions are as described in the table in https://github.com/phetsims/gravity-force-lab-basics/issues/182#issuecomment-541169252

**Comment by zepumph** [comment]
There were two bugs that I just committed fixes for on master and in the branch. This issue can still be closed.

**Comment by lmulhall-phet** [comment]
Seems to be working as described in https://github.com/phetsims/gravity-force-lab-basics/issues/182#issuecomment-541169252 in https://github.com/phetsims/QA/issues/438.
